### PR TITLE
Updated SharingClient libraries with latest version from HoloToolkit.

### PR DESCRIPTION
**Pull request by alexdrenea**
Enables server auto-discovery.

### CODE REVIEW

File: Assets/HoloToolkit/Sharing/Scripts/SDK/SystemRole.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Version 3.0.2
+ *
+ * Do not make changes to this file unless you know what you are doing--modify
+ * the SWIG interface file instead.
+ * ----------------------------------------------------------------------------- */
+
+namespace HoloToolkit.Sharing {
+
+public enum SystemRole {
+  SessionDiscoveryServerRole = 0,
```
Comment (NeerajW) [inline Assets/HoloToolkit/Sharing/Scripts/SDK/SystemRole.cs]: Did the default change for the entry? I think the previous default was Primary Client?

Comment (alexdrenea) [inline Assets/HoloToolkit/Sharing/Scripts/SDK/SystemRole.cs]: This is different than "HoloToolkit.Sharing.ClientRole" which is used in the client to specify Primary Client or Secondary Client from the user perspective. That remained unchanged with Primary Client being default.
This enum is used by the sharing framework to respond to a Network Discovery broadcast and I assume there is an internal mapping for the Client roles.

Comment (NeerajW) [inline Assets/HoloToolkit/Sharing/Scripts/SDK/SystemRole.cs]: That's right. Thanks for reminding me Alex.

**Comment by NeerajW** [comment]
@alexdrenea do you think a test scene with readme updated on the usage will help? I think it can also help clarify the nuances of the server selection etc.

**Comment by alexdrenea** [comment]
Interesting idea. As it stands right now, this PR does not provide any new functionality out-of-the-box, it simply enables one to write an auto-discover method to the SharingStage or a separate script. So maybe, rather than changing the SharingStage script and risking breaking already existing functionality, we could expose this as a new script - "Auto Discover Server" and present the script in a new (or existing) test scene and readme update. 
I'm still struggling to come up with a good option to allow user to pick from a list when multiple servers are found on the network since there are a lot of dependencies to consider for any kind of UI / interaction. I am open to any kind of suggestion or advice on that front.

**Comment by NeerajW** [comment]
@alexdrenea sounds good. Will merge this and if you could please open another issue tracking a new script for adding auto discovery.

**Comment by alexdrenea** [comment]
Created #172 .Sorry for the delay.

**Pull request merged**